src/EnergyPlus/ExhaustAirSystemManager.cc

```diff
@@ -0,0 +1,966 @@
+            Real64 FlowFrac = 0.0;
+            if (thisExhCtrl.MinExhFlowFracScheduleNum > 0) {
+                FlowFrac = EnergyPlus::ScheduleManager::GetCurrentScheduleValue(state, thisExhCtrl.ExhaustFlowFractionScheduleNum);
+            }
+
+            Real64 MinFlowFrac = 0.0;
+            if (thisExhCtrl.MinExhFlowFracScheduleNum > 0) {
+                MinFlowFrac = EnergyPlus::ScheduleManager::GetCurrentScheduleValue(state, thisExhCtrl.MinExhFlowFracScheduleNum);
+            }
+
+            if (FlowFrac < MinFlowFrac) {
+                FlowFrac = MinFlowFrac;
+            }
+
+            bool runExhaust = true;
+            if (thisExhCtrl.AvailScheduleNum == 0 ||
+                (thisExhCtrl.AvailScheduleNum > 0 &&
+                 EnergyPlus::ScheduleManager::GetCurrentScheduleValue(state, thisExhCtrl.AvailScheduleNum) > 0.0)) { // available
+                if (thisExhCtrl.MinZoneTempLimitScheduleNum > 0) {
+                    if (Tin >= EnergyPlus::ScheduleManager::GetCurrentScheduleValue(state, thisExhCtrl.MinZoneTempLimitScheduleNum)) {
+                        runExhaust = true;
+                    } else {
+                        runExhaust = false;
+                        FlowFrac = MinFlowFrac;
+                    }
+                } else {
+                    runExhaust = true;
+                    // flow not changed
+                }
+            } else {
+                runExhaust = false;
+                FlowFrac = 0.0; // directly set flow rate to zero.
+            }
+
+            if (thisExhCtrl.FlowControlTypeNum == 0) { // scheduled
+                MassFlow = DesignFlowRate * FlowFrac;
+            } else { // follow-supply
+                Real64 supplyFlowRate = 0.0;
+                int numOfSuppNodes = thisExhCtrl.SuppNodeNums.size();
+                for (int i = 1; i <= numOfSuppNodes; ++i) {
+                    supplyFlowRate += state.dataLoopNodes->Node(thisExhCtrl.SuppNodeNums(i)).MassFlowRate;
+                }
+                MassFlow = supplyFlowRate * FlowFrac;
+            }
+
+            if (thisExhCtrl.BalancedExhFracScheduleNum > 0) {
+                thisExhCtrl.BalancedFlow = // state.dataHVACGlobal->BalancedExhMassFlow =
+                    MassFlow *             // state.dataHVACGlobal->UnbalExhMassFlow *
+                    EnergyPlus::ScheduleManager::GetCurrentScheduleValue(state, thisExhCtrl.BalancedExhFracScheduleNum);
+                thisExhCtrl.UnbalancedFlow =  // state.dataHVACGlobal->UnbalExhMassFlow =
+                    MassFlow -                // = state.dataHVACGlobal->UnbalExhMassFlow -
+                    thisExhCtrl.BalancedFlow; // state.dataHVACGlobal->BalancedExhMassFlow;
+            } else {
+                thisExhCtrl.BalancedFlow = 0.0;
+                thisExhCtrl.UnbalancedFlow = MassFlow;
+            }
+
+            state.dataLoopNodes->Node(InletNode).MassFlowRate = MassFlow;
+        }
+
```

Comment:
Can create `outletNode` and `inletNode` shortcut references here.

Reply:
The reference short cuts are now created to reduced code length.